PSKoans/Koans/IntroTraining/AboutInts.koans.ps1

```diff
@@ -0,0 +1,65 @@
+using module PSKoans
+[Koan(Position = 110)]
+param()
+<#
+    INTs
```

Comment:
I like the explanations in this file, but I'm not sure this is information that a beginner really needs...

PowerShell plays pretty fast and loose with types in general, and even more so with numbers. You won't really hit overflows or underflows very often, so you generally don't need to _care_ what type a number actually is in a majority of cases.

These explanations would make a lot more sense in the context of the [bitwise operations](https://github.com/vexx32/PSKoans/blob/master/PSKoans/Koans/Constructs%20and%20Patterns/AboutBitwiseOperations.Koans.ps1) file where @indented-automation went into more detail on how these types are built up from individual bits and bytes.

I'm thinking this is a bit out of place in the introductory segment here. Potentially we should cover the basic number types in general, not just integers, and maybe go into how PowerShell interprets numbers by default (e.g., 10 is integer, 10.0 is double, and anything over int range goes into int64 automatically), but I don't think we should go into this level of detail in such an introductory topic.

Reply:
I'd say it's important to understand, at least at a basic level, a wide-range of datatypes when learning any new language. Strings, INTs and Booleans seemed to me the absolutely bare minimum required. Then once individuals are used to that introduction to Arrays, Hashtables and Functions becomes that must easier.

I can see that the bitwise file does a pretty good job expanding on binary to cover integers in detail 🦄 

We could replace AboutINTs with AboutNumbers and cover:
- Integers
- Doubles

With a footnote referencing AboutBitwiseOperations for further details?

And/or convert the whole thing into an advise file?

Reply:
I'd say it's worth covering it as AboutNumbers and sticking to the more basic & practical side of things, yep. A mention of the future topic would be good too! :blush:

Reply:
I couldn't quite work out how to do a Koan on just integers and doubles, so I've made an advise file instead:

```
PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans> Show-Advice -Name Numbers.Advise

Those who came before offer you a fragment of wisdom.

     Numbers

   In PowerShell, we can use many different data types to represent numbers. All differ slightly in what they can and can't
   do. The two most common ones are:

        Intergers (INT)
                Represent whole numbers. This is the default numeric data type that PowerShell uses

        Doubles
                Represents numbers with decimal places

   So 10 will be an integer and 10.0 will be a double. Why does this matter? If you perform any operation with a double,
   then the result of that will be a double:

        PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans> 10 * 10.0
                100
        PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans> 10 * 10.0 | Get-Member


        TypeName: System.Double

   Say we're performing a very specific calculation. Lets say Pi. We'be worked out a portion of Pi being equal to
   3.14159265358979, what happens if we pass this value into a function that excepts an integer as input?

        PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans> function HelloWorld([int32]$Int) {$Int}
        PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans> $Pi = 3.14159265358979
        PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans> HelloWorld -Int $Pi
        3

   Our number is rounded to the nearest integer, and suddenly whatever calculation we're performing is a little less
   accurate.

PS C:\Users\Administrator\Documents\GitHub\Dev\PSKoans>
```

Also is there a reason you need to do the .advise on the end of the topic name? Might be more user friendly if you could just type in the name, and/or have a -list param to show all the advise topics available? Happy to open an issue for it if you feel like it's a legitimate point.

This does then bring in to question why the AboutBinary file is in the position that it is, given that it was originally there as a build-up to INTs.

Reply:
Yeah that cmdlet could use an enhancement to only need to specify the name without the `.Advice` portion. List param is also a good idea.

I'm not sure an advice file is the best fit here either, to be honest... This does kind of need to be a koan file, just because of the necessary length. You have the right idea for it here already, just need to frame it so that the answers are something the user finds out rather than is directly given. To take one of your examples:

```powershell
Describe 'Basic Number Types' {
    Context 'Double' {
        It 'can result from an operation involving multiple types of numbers' {
            $Int = 10
            $Double = 10.0
            'System.____' | Should -Be $Int.GetType().Fullname
            'System.____' | Should -Be $Double.GetType().Fullname

            $Result = $Int * $Double
            # What type results when you multiply integers and doubles?
            'System.____' | Should -Be $Result.GetType().Fullname
        }
    }
}
```

Your others are similar, the example with pi is a great one -- you can actually borrow `[math]::PI` for that example to be very explicit about it and show clearly how the rounding works when casting to int, etc.

Though it's a slightly more advanced thing, it _may_ be worth briefly covering how that rounding functions, since it can and does directly affect input to commands sometimes; PS uses Banker's Rounding (2.5 rounds to 2, but 3.5 rounds to 4; basically if it's .5 and you convert it to int, it'll round to the nearest even number instead of just rounding up all the time.

It doesn't necessarily need to be a long or super comprehensive file, but it's good to have these examples shown early so folks are somewhat aware that there are multiple number types common to PS and how they can be interacted with. Full examination of both types can come in a more advanced topic. 🙂

Reply:
I think I get what you're saying. See the new commit and let me know what you think.

Reply:
@vexx32 heya! Any update? Keen to get this merged 🦄

Reply:
Ah, sorry! Sidetracked.

This looks good! I've a few comments to make and then I'll publish that review, and then we're just about done. 😊